### DISCUSSION TITLE
AGGREGATE 模型随着ETL不断进行，查询结果不符合预期

### DISCUSSION THREAD

**Discussion by linlin994395** [apache/doris#22112] [A - General / Q&A]
**1. 建表语句**
create table studio_xxx.xxx_dim_server
(
    `server` int COMMENT '服务器',
    `pt_dt` date min NOT NULL COMMENT '开服日期'
) ENGINE=OLAP
AGGREGATE KEY(`server`) COMMENT 'OLAP'
DISTRIBUTED BY HASH(`server`) BUCKETS 1
PROPERTIES (
    "replication_num" = "1"
);
**2. 初始化插入语句**
INSERT INTO studio_xxx.xxx_dim_server
select
replace(jsonb_extract(data, '$.server'), '"', ""), min(pt_dt) pt_dt
from studio_xxx.xxx_ods_total
where event_name in ('LogReg', 'LogLogin', 'LogPay')
group by replace(jsonb_extract(data, '$.server'), '"', "")
初始化插入后结果如下
![image](https://github.com/apache/doris/assets/22185640/89f2b87f-54fd-4234-9d3e-4a02550adff8)
**3. ETL语句 每10分钟执行一次**
INSERT INTO studio_xxx.xxx_dim_server
select
replace(jsonb_extract(data, '$.server'), '"', "") server,
min(pt_dt) pt_dt
from studio_xxx.xxx_ods_total
where pt_dt >= '{start_time}' and pt_dt <= '{end_time}'
and event_name in ('LogReg', 'LogLogin', 'LogPay')
group by replace(jsonb_extract(data, '$.server'), '"', "") 
**4. 过一段时间 select结果与预期不符（原有2服，3服数据丢失，1服数据pt_dt发生非预期改变）**
![image](https://github.com/apache/doris/assets/22185640/b79bed1f-2315-4763-a8f9-2fbd92435202)

**分析：**
1. 怀疑是多副本原因，每次查询到不同的BE节点，BE节点间同步不及时。但设置为单副本，问题依旧，故排除
2. 怀疑AGGREGATE模型在min，replace等聚合方式下，聚合有问题，求解决

**Comment by LemonLiTree** [comment]
which version？

**Reply by linlin994395** [reply]
doris-1.2.2-rc01

**Reply by LemonLiTree** [reply]
show variables like '%vec%';

**Reply by linlin994395** [reply]
![image](https://github.com/apache/doris/assets/22185640/b54f826e-abb0-474f-90d3-00f6134d5ae7)
![image](https://github.com/apache/doris/assets/22185640/29749de9-ed68-4249-89d8-12e996ac1394)
